app/code/core/Mage/Admin/Model/Roles.php

```diff
@@ -152,24 +152,30 @@ protected function _buildResourcesArray(
             $level = -1;
         } else {
             $resourceName = $parentName;
-            if (!in_array($resource->getName(), array('title', 'sort_order', 'children', 'disabled'))) {
+            if (!in_array($resource->getName(), array('title', 'sort_order', 'children', 'disabled', 'acl'))) {
                 $resourceName = (is_null($parentName) ? '' : $parentName . '/') . $resource->getName();
 
                 //assigning module for its' children nodes
                 if ($resource->getAttribute('module')) {
                     $module = (string)$resource->getAttribute('module');
                 }
 
+                if ($resource->acl) {
+                    $aclpath = (string)$resource->acl;
+                } else {
+                    $aclpath = $resourceName;
+                }
+
                 if ($rawNodes) {
-                    $resource->addAttribute("aclpath", $resourceName);
+                    $resource->addAttribute("aclpath", $aclpath);
                     $resource->addAttribute("module_c", $module);
                 }
 
                 if (is_null($represent2Darray)) {
-                    $result[$resourceName]['name']  = Mage::helper($module)->__((string)$resource->title);
-                    $result[$resourceName]['level'] = $level;
+                    $result[$aclpath]['name']  = Mage::helper($module)->__((string)$resource->title);
+                    $result[$aclpath]['level'] = $level;
```

Comment:
I am not positive it's okay to change the array key here. Before I had something like:
```php
$result[$resourceName]['name']  = Mage::helper($module)->__((string)$resource->title);
$result[$resourceName]['level'] = $level;
$result[$resourceName]['aclpath'] = $aclpath; // New array value
```

But that will require more code changes in core. For example, in the admin role edit screen: https://github.com/OpenMage/magento-lts/blob/0d997174c676cdb154162bd945f75bb37bb0b2b2/app/code/core/Mage/Adminhtml/Block/Permissions/Tab/Rolesedit.php#L103-L111

We only check `array_key_exists`, but would need to also check for `aclpath` sub-array key.